packages/solidity-contracts/contracts/fuelchain/FuelMessagePortal/v3/FuelMessagePortalV3.sol

```diff
@@ -9,13 +9,43 @@ contract FuelMessagePortalV3 is FuelMessagePortalV2 {
     using FuelBlockHeaderLiteLib for FuelBlockHeaderLite;
 
     error MessageBlacklisted();
-    error WithdrawalsPaused();
     error MessageRelayFailed();
+    error RateLimitExceeded();
+    error WithdrawalsPaused();
+
+    /// @dev The rate limit setter role
+    bytes32 public constant SET_RATE_LIMITER_ROLE = keccak256("SET_RATE_LIMITER_ROLE");
+
+    /// @notice Duration after which rate limit resets.
+    uint256 public immutable rateLimitDuration;
+
+    /// @notice Amounts already withdrawn this period.
+    uint256 public currentPeriodAmount;
 
+    /// @notice Flag to indicate whether withdrawals are paused or not.
     bool public withdrawalsPaused;
+
+    /// @notice The time at which the current period ends at.
+    uint256 public currentPeriodEnd;
+
+    /// @notice The eth withdrawal limit amount.
+    uint256 public limitAmount;
+
     mapping(bytes32 => bool) public messageIsBlacklisted;
 
-    constructor(uint256 _depositLimitGlobal) FuelMessagePortalV2(_depositLimitGlobal) {}
+    constructor(uint256 _depositLimitGlobal, uint256 _rateLimitDuration) FuelMessagePortalV2(_depositLimitGlobal) {
+        rateLimitDuration = _rateLimitDuration;
+    }
+
+    function initializerV3(FuelChainState fuelChainState, uint256 _limitAmount) public initializer {
+        super.initialize(fuelChainState);
```

Comment:
let's be specific about what `initialize` function is being called, use `FuelMessagePortal.initialize()`